src/utils.rs

```diff
@@ -56,17 +62,48 @@ impl<C: FirebirdClient> SystemInfos for Connection<C> {
     }
 
     fn server_engine(&mut self) -> Result<EngineVersion, FbError> {
-        todo!()
+        let row: Option<(String,)> = self.query_first(
+            "SELECT rdb$get_context('SYSTEM', 'ENGINE_VERSION') from rdb$database;",
+            (),
+        )?;
+
+        if let Some((version,)) = row {
+            return match version {
+                ver if ver.starts_with("4.") => Ok(EngineVersion::V4),
+                ver if ver.starts_with("3.") => Ok(EngineVersion::V3),
+                ver if ver.starts_with("2.") => Ok(EngineVersion::V2),
+                ver => Err(FbError::from(format!("Version not detected: {}", ver)))
```

Comment:
little bit shorter.

```suggestion
            return match &version[0..2] {
                "4." => Ok(EngineVersion::V4),
                "3." => Ok(EngineVersion::V3),
                "2." => Ok(EngineVersion::V2),
                _ => Err(FbError::from(format!("Version not detected: {}", version)))
```

Reply:
Just to avoid panics, it's better to use `&version.get(0..2)` in the match instead:

```rust
            return match &version.get(0..2) {
                Some("4.") => Ok(EngineVersion::V4),
                Some("3.") => Ok(EngineVersion::V3),
                Some("2.") => Ok(EngineVersion::V2),
                _ => Err(FbError::from(format!("Version not detected: {}", version)))
```

Reply:
Done!